Sources/TestingMacros/Support/DiagnosticMessage.swift

```diff
@@ -643,6 +643,22 @@ struct DiagnosticMessage: SwiftDiagnostics.DiagnosticMessage {
     )
   }
 
+  /// Create a diagnostic message stating that a string literal expression
+  /// passed as the display name to a `@Test` or `@Suite` attribute is empty
+  /// but should not be.
+  ///
+  /// - Parameters:
+  ///   - displayNameExpr: The display name string literal expression.
+  ///
+  /// - Returns: A diagnostic message.
+  static func emptyDisplayName(_ displayNameExpr: StringLiteralExprSyntax) -> Self {
+    Self(
+      syntax: Syntax(displayNameExpr),
+      message: "Display name string should not be empty",
+      severity: .warning
+    )
```

Comment:
Should be easy enough to provide a fix-it to remove the empty display name. Bonus points for offering a second fix-it to turn it into a placeholder and force the test author to write something.